add bitlayer fees adapter

stromman opened the pull request:
we creat a adapter for bitlayer project
PLS check and fallback for fees/bitlayer/index.ts

dtmkeng commented:
could you resolve conflicts

llamatester commented:
The bitlayer adapter exports: 
        
 
 _______________________________________
    
    
    
    ERROR: Use Adapter v2 format for new adapters

dtmkeng commented on fees/bitlayer/index.ts:
```diff
@@ -0,0 +1,91 @@
+}
+
+const adapter: Adapter = {
+  adapter: {
```
add `version: 2` above `adapter`

dtmkeng commented on fees/bitlayer/index.ts:
```diff
@@ -0,0 +1,91 @@
+          "https://api.btrscan.com/scan/v1/chain/txForDefillama"
+        );
+
+        console.log("amount", amount);
```
remove console log

dtmkeng commented on fees/bitlayer/index.ts:
```diff
@@ -0,0 +1,91 @@
+          timestamp: options.startOfDay,
+          dailyFees,
+        };
+      },
```
add `runAtCurrTime: true` above `start`

llamatester commented:
The bitlayer adapter exports: 
        
 
 _______________________________________
    Fees for 7/5/2024
    _______________________________________
    
    TypeError: Cannot destructure property 'startOfDay' of 'undefined' as it is undefined.
        at getEtherscanFees (/home/runner/work/dimension-adapters/dimension-adapters/fees/bitlayer/index.ts:40:5)
        at fetch (/home/runner/work/dimension-adapters/dimension-adapters/fees/bitlayer/index.ts:68:30)
        at getChainResult (/home/runner/work/dimension-adapters/dimension-adapters/adapters/utils/runAdapter.ts:31:25)
        at async Promise.all (index 0)
        at async /home/runner/work/dimension-adapters/dimension-adapters/cli/testAdapter.ts:75:21 {
      chain: 'bitlayer'
    }

llamatester commented:
The bitlayer adapter exports: 
        
 
 _______________________________________
    Fees for 7/5/2024
    _______________________________________
    
    TypeError: Cannot destructure property 'startOfDay' of 'undefined' as it is undefined.
        at getEtherscanFees (/home/runner/work/dimension-adapters/dimension-adapters/fees/bitlayer/index.ts:40:5)
        at fetch (/home/runner/work/dimension-adapters/dimension-adapters/fees/bitlayer/index.ts:68:30)
        at getChainResult (/home/runner/work/dimension-adapters/dimension-adapters/adapters/utils/runAdapter.ts:31:25)
        at async Promise.all (index 0)
        at async /home/runner/work/dimension-adapters/dimension-adapters/cli/testAdapter.ts:75:21 {
      chain: 'bitlayer'
    }

llamatester commented:
The bitlayer adapter exports: 
        
 
 _______________________________________
    Fees for 7/5/2024
    _______________________________________
    
    TypeError: Cannot destructure property 'startOfDay' of 'undefined' as it is undefined.
        at getEtherscanFees (/home/runner/work/dimension-adapters/dimension-adapters/fees/bitlayer/index.ts:40:5)
        at fetch (/home/runner/work/dimension-adapters/dimension-adapters/fees/bitlayer/index.ts:68:30)
        at getChainResult (/home/runner/work/dimension-adapters/dimension-adapters/adapters/utils/runAdapter.ts:31:25)
        at async Promise.all (index 0)
        at async /home/runner/work/dimension-adapters/dimension-adapters/cli/testAdapter.ts:75:21 {
      chain: 'bitlayer'
    }

dtmkeng commented on fees/bitlayer/index.ts:
```diff
@@ -0,0 +1,90 @@
+      fetch: async (
+        _timestamp: number,
+        _: ChainBlocks,
+        options: FetchOptions
```
param should only (options: FetchOptions)

llamatester commented:
The bitlayer adapter exports: 
        
 
 _______________________________________
    Fees for 7/5/2024
    _______________________________________
    
    [Error: Request failed with status code 429] {
      url: 'https://api.btrscan.com/scan/v1/chain/txForDefillama',
      method: 'POST',
      axiosError: 'Too Many Requests',
      chain: 'bitlayer'
    }

dtmkeng commented:
seem api error

stromman commented:
> seem api error
Frequency Limit

llamatester commented:
The bitlayer adapter exports: 
        
 
 _______________________________________
    Fees for 7/5/2024
    _______________________________________
    
    [Error: Request failed with status code 429] {
      url: 'https://api.btrscan.com/scan/v1/chain/txForDefillama',
      method: 'POST',
      axiosError: 'Too Many Requests',
      chain: 'bitlayer'
    }

dtmkeng commented:
still error could you check

llamatester commented:
The bitlayer.ts adapter exports: 
        
 
 _______________________________________
    
    
    
    ERROR: Use Adapter v2 format for new adapters

llamatester commented:
The bitlayer adapter exports: 
        
 
 _______________________________________
    
    
    
    ERROR: Use Adapter v2 format for new adapters

llamatester commented:
The bitlayer adapter exports: 
        
 
 _______________________________________
    Fees for 7/5/2024
    _______________________________________
    
    BITLAYER 👇
    Backfill start time: 14/4/2024
    NO METHODOLOGY SPECIFIED
    Timestamp: 1715094481 (2024-05-07T15:08:01.000Z)
    Daily fees: 0

dtmkeng commented:
is there probby some day return daily fees is 0  ?

stromman commented:
> is there probby some day return daily fees is 0 ?

nop, something wrong.repairing

AoWangg commented:
@dtmkeng Dear reviewer, please re-examine this code as we have addressed the issues previously pointed out.

dtmkeng commented:
> @dtmkeng Dear reviewer, please re-examine this code as we have addressed the issues previously pointed out. Thank you very much.

it is this web https://www.bitlayer.org/gas-swap ?

stromman commented:
> Dear reviewer, please re-examine this code as we have addressed the issues previously pointed out. Thank you very much.

I thank that you mean : https://www.btrscan.com/

AoWangg commented:
> > @dtmkeng Dear reviewer, please re-examine this code as we have addressed the issues previously pointed out. Thank you very much.
> 
> it is this web https://www.bitlayer.org/gas-swap ?

Thanks for the merge.

Here is some information about Bitlayer

**Website:** https://www.bitlayer.org/
**Twitter:** https://twitter.com/BitLayerLabs

llamatester commented:
The bitlayer adapter exports: 
        
 
 _______________________________________
    Fees for 9/5/2024
    _______________________________________
    
    BITLAYER 👇
    Backfill start time: 14/4/2024
    NO METHODOLOGY SPECIFIED
    Timestamp: 1715262340 (2024-05-09T13:45:40.000Z)
    Daily fees: 5.37 k

dtmkeng commented:
hmm why i call at my local it error 
```
[Error: connect ETIMEDOUT 2606:4700:20::681a:4ac:443] {
  url: 'https://api.btrscan.com/scan/v1/chain/txForDefillama',
  method: 'POST',
  chain: 'bitlayer'
}
```
